tokio-util/src/sync/tests/mpsc.rs

```diff
@@ -0,0 +1,26 @@
+// TODO: loom
+use crate::sync::Sender;
+use futures::future::poll_fn;
+use tokio::sync::mpsc::channel;
+
+fn block_on<F: std::future::Future>(fut: F) -> F::Output {
+    tokio::pin!(fut);
+    let mut cx = std::task::Context::from_waker(futures::task::noop_waker_ref());
+    loop {
+        if let std::task::Poll::Ready(r) = fut.as_mut().poll(&mut cx) {
+            return r;
+        }
+    }
+}
```

Comment:
Why not just test it with `futures::executor::block_on`?

Reply:
I was afraid that `futures::executor::block_on` doesn't work under MIRI because it uses thread parking stuff.
But now I see that whole futures stuff does not support MIRI, so manual `block_on` reimplementation is not needed anymore.